File: ru102j/redisolar/src/main/java/com/redislabs/university/RU102J/dao/MetricDao.java
package com.redislabs.university.RU102J.dao;

import com.redislabs.university.RU102J.api.Measurement;
import com.redislabs.university.RU102J.api.MeterReading;
import com.redislabs.university.RU102J.api.ValueUnit;

import java.util.List;

public interface MetricDao {
    public void insert(MeterReading reading);
    public List<Measurement> getRecent(Long siteId, ValueUnit unit, Integer limit);
}

File: ru102js/redisolar/daos/metric_dao.js
const daoLoader = require('./daoloader');

const impl = daoLoader.loadDao('metric');

const insert = meterReading => impl.insert(meterReading);

const getRecent = (siteId, unit, time, limit) => impl.getRecent(siteId, unit, time, limit);

module.exports = {
  insert,
  getRecent,
};
